### PR TITLE
Add support for processing Template Kit ZIP files.

### PR DESCRIPTION
**Pull request by dtbaker**
## PR Checklist

- [x] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

* Add support for importing Template Kit ZIP files

## Description

Template Kit ZIP files contain a main `manifest.json` file and a folder of json `templates/` like this:

image

When trying to upload this to Elementor it returns a generic error:

image

This is because the code doesn't support unzipping a file with nested folders of json templates, and it tries to parse the main `manifest.json` file as an Elementor template (it's not).

## Changes

- Replace the single directory level `scandir` with a new recursive file finder.
- Ignore files named `manifest.json` in the ZIP 
- This allows the ZIP import process to find templates nested in temporary extracted folders.

## Why?

We would like to add basic support for Template Kit ZIP files into Elementor core, rather than making users download the separate [Template Kit Import](https://wordpress.org/plugins/template-kit-import/) plugin each time. This change is a step in the right direction to making Elementor core support Template Kit ZIP format. 

## Test instructions

* ZIP up some Elementor templates, but put them into a `templates/` folder in the ZIP file
* Upload ZIP file to Elementor as usual
* Check they all import

## Quality assurance

- [x] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: includes/template-library/sources/local.php [hunk 1]
```diff
@@ -777,6 +777,52 @@ public function export_multiple_templates( array $template_ids ) {
...
 
+	/**
+	 * Find temporary files.
+	 *
+	 * Recursively finds a list of temporary files from the extracted zip file.
+	 *
+	 * Example return data:
+	 *
+	 * [
+	 *  0 => '/www/wp-content/uploads/elementor/tmp/5eb3a7a411d44/templates/block-2-col-marble-title.json',
```
Comment (matipojo) [inline includes/template-library/sources/local.php:788]: Two rows are enough.

File: includes/template-library/sources/local.php [hunk 2]
```diff
@@ -818,7 +864,8 @@ public function import_template( $name, $path ) {
 			for ( $i = 0; $i < $zip->numFiles; $i++ ) {
 				$zipped_file_name = $zip->getNameIndex( $i );
 				$zipped_extension = pathinfo( $zipped_file_name, PATHINFO_EXTENSION );
-				if ( 'json' === $zipped_extension ) {
+				// Template Kit zip files contain a `manifest.json` file, this is not a valid Elementor template so ensure we skip it.
+				if ( 'manifest.json' !== $zipped_file_name && 'json' === $zipped_extension ) {
```
Comment (matipojo) [inline includes/template-library/sources/local.php:868]: Flip the conditions, the `manifest.json` should be checked only for json files.

**Review by matipojo — CHANGES_REQUESTED** [review]

**Comment by shilo-ey** [comment]
@dtbaker Have you gotten the notes?

**Comment by shilo-ey** [comment]
@dtbaker please send this PR to develop branch

Thanks

File: includes/template-library/sources/local.php (hunk 2 above)
Comment (dtbaker) [reply includes/template-library/sources/local.php]: Thanks @matipojo - feedback addressed :) https://github.com/elementor/elementor/pull/11316/commits/afefd3c557d0b6b0236d5906a5b6ac52d6e4012c

File: includes/template-library/sources/local.php (hunk 1 above)
Comment (dtbaker) [reply includes/template-library/sources/local.php:788]: Thanks @matipojo - feedback addressed :) https://github.com/elementor/elementor/pull/11316/commits/afefd3c557d0b6b0236d5906a5b6ac52d6e4012c

**Comment by dtbaker** [comment]
@shilo-ey I've updated to `develop` :+1:

**Review by matipojo — APPROVED** [review]

**Review by kobizz — APPROVED** [review]

**Review by KingYes — APPROVED** [review]

**Pull request merged**